packages/website/src/components/layout/EditorTabs.tsx

```diff
@@ -0,0 +1,55 @@
+import EditIcon from '@site/src/icons/edit.svg';
+import clsx from 'clsx';
+import React from 'react';
+
+import styles from './EditorTabs.module.css';
+
+export interface EditorTabsProps<T extends string | boolean> {
+  readonly tabs: ({ value: T; label: string } | T)[];
+  readonly active: T;
+  readonly showVisualEditor?: boolean;
+  readonly change: (name: T) => void;
+  readonly showModal?: (name: T) => void;
```

Comment:
Out of curiosity, is there any benefit to marking props as `readonly`? It feels like extra effort - maybe we should use a lint rule against modifying props instead?

Reply:
> maybe we should use a lint rule against modifying props instead?

that's what readonly is for, adding rule that covers buildin behavior is bad

there is actually a rule to mark props as readonly (for flow) https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/prefer-read-only-props.md